controllers/complianceeventsapi/server.go

```diff
@@ -738,6 +789,20 @@ func postComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	authorized, err := getWriteAuth(config, r.Header.Get("Authorization"), reqEvent.Cluster.Name)
```

Comment:
Could you pass the context from `r.Context()` to this method for API requests it makes instead of using `context.TODO()`? Same for the get API request.